packages/toolpad-components/src/DataGrid.tsx

```diff
@@ -289,7 +292,35 @@ const DataGridComponent = React.forwardRef(function DataGridComponent(
   }: ToolpadDataGridProps,
   ref: React.ForwardedRef<HTMLDivElement>,
 ) {
+  const components = useComponents();
+
   const nodeRuntime = useNode<ToolpadDataGridProps>();
+  const columnTypes = React.useMemo(
+    () => ({
+      ...CUSTOM_COLUMN_TYPES,
+      codeComponent: {
+        renderCell: (params: GridRenderCellParams) => {
+          const { value, colDef, row, field } = params;
+          const column = colDef as SerializableGridColumn;
+
+          invariant(components, 'Components must be defined');
```

Comment:
Why wouldn't this be defined? We use `createProvidedContext`

Reply:
You're right, I think it might be a leftover from previous implementation when TS complained about possibility of undefined components, but that's certainly incorrect. Removed 👍